isodatetime/data.py

```diff
@@ -1111,6 +1113,32 @@ def get_largest_truncated_property_name(self):
                 return attr
         return None
 
+    def get_smallest_missing_property_name(self):
+        """Return the smallest unit missing
+        from a truncated representation."""
+        if not self.truncated:
+            return None
+        prop_dict = self.get_truncated_properties()
+        attr_keys = ["year_of_century", "decade_of_century",
+                     "year_of_decade", "month_of_year",
+                     "week_of_year", "day_of_year", "day_of_month",
+                     "day_of_week", "hour_of_day", "minute_of_hour",
+                     "second_of_minute"]
+        attr_dict = {"year_of_century": "century",
```

Comment:
Minor matter, defining the keys and dictionary separately is a little clunky.

The simplest option is probably to do something like:

```python
attr_key_map = [
    ('year_of_century', 'century'),
    ('year_of_decade', 'decade_of_century'),
    ...
]
for attr, name in attr_key_map:
    if attr in prop_dict:
        return name
```